src/components/LoginBox.tsx

```diff
@@ -83,41 +78,44 @@ function LoginBox(): JSX.Element {
     if (!loginError) logIn(name);
   }
 
-  // https://mui.com/components/text-fields/
-  return (
-    <LoginRoot>
-      <LoginContainer>
-        <form className="loginForm" noValidate autoComplete="off">
-          <h1>LOG IN</h1>
-          <LoginFields>
-            <TextField
-              required
-              error={nameErr !== ""}
-              id="username-input"
-              label="Username"
-              helperText={nameErr}
-              onChange={(e) => setName(e.target.value)}
-            />
-            <br />
-            <TextField
-              required
-              error={passErr !== ""}
-              id="password-input"
-              label="Password"
-              type="password"
-              autoComplete="current-password"
-              helperText={passErr}
-              onChange={(e) => setPass(e.target.value)}
-            />
-            <br />
-            <LoginButton variant="contained" onClick={() => ValidateLogin()}>
-              Log In
-            </LoginButton>
-          </LoginFields>
-        </form>
-      </LoginContainer>
-    </LoginRoot>
-  );
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        // Retreive the name and password manually
+        // Normally we could just use states and update user/pass on change, but that doesn't work with autofill
+        ValidateLogin(event.target.elements["username-input"]._valueTracker.getValue(), event.target.elements["password-input"]._valueTracker.getValue());
+      };
```

Comment:
If it is possible to keep the loginBox simple with only the return DOM will be grate.
Personally I'll extract this function and put in the util folder. Which also allow for better testing.

Reply:
I've been reading your comment and I'm not sure how I'd make it only return the DOM with no functions.
I can simplify it by making it 1 function instead of 2, but there needs to be a handleSubmit() that takes an event and makes changes to the error messages that are displayed.

Reply:
it's simply  by creating a custom hook. And imported. You can refer to the code that i wrote in overlayfunc. useOverlay. and how I imported useOverlay in overlay.tsx